test/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/EISConnectorSpec.scala

```diff
@@ -0,0 +1,140 @@
+               |            "recordId": "8ebb6b04-6ab0-4fe2-ad62-e6389a8a204f",
+               |            "traderRef": "BAN001001",
+               |            "comcode": "104101000",
+               |            "accreditationRequest": "Not requested",
+               |            "goodsDescription": "Organic bananas",
+               |            "countryOfOrigin": "EC",
+               |            "category": 3,
+               |            "assessments": [
+               |                {
+               |                    "assessmentId": "abc123",
+               |                    "primaryCategory": "1",
+               |                    "condition": {
+               |                        "type": "abc123",
+               |                        "conditionId": "Y923",
+               |                        "conditionDescription": "Products not considered as waste according to Regulation (EC) No 1013/2006 as retained in UK law",
+               |                        "conditionTraderText": "Excluded product"
+               |                    }
+               |                }
+               |            ],
+               |            "supplementaryUnit": 500,
+               |            "measurementUnit": "square meters(m^2)",
+               |            "comcodeEffectiveFromDate": "2024-11-18T23:20:19Z",
+               |            "comcodeEffectiveToDate": "",
+               |            "version": 1,
+               |            "active": true,
+               |            "toReview": false,
+               |            "reviewReason": null,
+               |            "declarable": "IMMI declarable",
+               |            "ukimsNumber": "XIUKIM47699357400020231115081800",
+               |            "nirmsNumber": "RMS-GB-123456",
+               |            "niphlNumber": "6 S12345",
+               |            "locked": false,
+               |            "srcSystemName": "CDAP",
+               |            "createdDateTime": "2024-11-18T23:20:19Z",
+               |            "updatedDateTime": "2024-11-18T23:20:19Z"
+               |        }
+               |    ],
+               |    "pagination": {
+               |        "totalRecords": 1,
+               |        "currentPage": 0,
+               |        "totalPages": 1,
+               |        "nextPage": null,
+               |        "prevPage": null
+               |    }
+               |}
+               |""".stripMargin)
+
+    val expectedResponse: GetEisRecordsResponse = sampleJson.as[GetEisRecordsResponse]
+    val httpResponse                            = mock[HttpResponse]
+
+    when(httpResponse.json).thenReturn(sampleJson)
+    when(httpResponse.status).thenReturn(200)
+
+    when(requestBuilder.execute[HttpResponse](any(), any())).thenReturn(Future.successful(httpResponse))
+
+    val result = eisConnector.fetchRecord(eori, recordId, correlationId)(ec, hc)
+
+    result.map { res =>
```

Comment:
This may make the test brittle, as we are not waiting on the feature and the test could finish before the feature return. This could cause the test some time not passing depending how fast the test run. I would use a wait

`

val result = await(eisConnector.fetchRecord(eori, recordId, correlationId)(ec, hc))
    
    result shouldBe expectedResponse

`